vpr/src/base/partition_region.cpp

```diff
@@ -48,6 +50,20 @@ PartitionRegion intersection(PartitionRegion& pr1, PartitionRegion& pr2) {
     return pr;
 }
 
```

Comment:
Let's talk about this in our meeting this week.  I can merge this to get it going, but this update_cluster_part_reg has a few problems:
1. name is overly specific; this is an "in-place" intersection, rather than having anything to do with clustering specifically. We should at least change the name to reflect what the routine does, rather than who calls it right now.
2. As written it is probably not faster than the other intersection routine, since it still creates a new partition_region and assigns it to the cluster_pr.  So if the other routine is leading to a negligible cluster time increase we can delete this one.  I know I requested an in-place version for speed, but looking at this I don't think it is "in place" enough to be faster.